rel/pattern_set.go

```diff
@@ -102,7 +91,11 @@ func (p SetPattern) Bind(ctx context.Context, local Scope, value Value) (context
 				return ctx, EmptyScope, fmt.Errorf("the length of set %s is wrong", set)
 			}
 
-			ctx, scope, err = p.patterns[i].Bind(ctx, local, set.set.Any().(Value))
+			e := set.Enumerator()
+			if !e.MoveNext() {
+				panic("set with count 1 failed to enumerate")
```

Comment:
is this an implementation error? if not, it should return the error.

Reply:
It is an implementation error. The count is checked a few lines prior.